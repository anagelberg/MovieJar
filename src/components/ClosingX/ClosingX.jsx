import './ClosingX.scss';
import { ReactComponent as CloseIcon } from '../../assets/icons/close.svg'

function ClosingX({ closeHandler, modifier }) {
    return <CloseIcon
        className={modifier
            ? `close-x close-x${modifier}`
            : "close-x"
        }
        onClick={closeHandler} />
}

export default ClosingX;