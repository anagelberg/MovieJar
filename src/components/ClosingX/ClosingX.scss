@use "../../styles/variables" as *;
@use "../../styles/mixins" as *;
@use "../../styles/animations" as *;

.close-x {
  fill: $dark-text;
  position: absolute;
  right: $icon-spacing-unit;
  top: $icon-spacing-unit;
  @include small-icon;

  &--light {
    fill: $light-text;
  }
}
