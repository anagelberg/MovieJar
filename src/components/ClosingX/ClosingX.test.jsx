import { render, screen } from '@testing-library/react';
import ClosingX from './ClosingX';
import userEvent from '@testing-library/user-event';

test('renders ClosingX component', () => {
    render(<ClosingX />);
    expect(screen.getByTestId('closing-x')).toBeInTheDocument();
});

test("ClosingX component matches snapshot", () => {
    const { container } = render(<ClosingX />);
    expect(container.firstChild).toMatchSnapshot();
});

test('toggles ClosingX modifier class', () => {
    const { container } = render(<ClosingX modifier="--testing-class" />);
    expect(container.firstChild).toHaveClass('close-x--testing-class')
});

test('calls closeHandler on click', () => {
    const closeHandler = jest.fn();
    render(<ClosingX closeHandler={closeHandler} />);
    userEvent.click(screen.getByTestId('closing-x'));
    expect(closeHandler).toHaveBeenCalled();
});