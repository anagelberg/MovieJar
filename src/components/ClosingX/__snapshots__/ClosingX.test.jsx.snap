// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClosingX component matches snapshot 1`] = `
<svg
  class="close-x"
  data-testid="closing-x"
>
  close.svg
</svg>
`;
